src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +46,605 @@ class Outer
+                // public class Derived2 : Base { } // 2, 3
+                Diagnostic(ErrorCode.ERR_BadVisBaseClass, "Derived2").WithArguments("Derived2", "Base").WithLocation(3, 14),
+                // (3,14): error CS9301: File type 'Base' cannot be used as a base type in non-file type 'Derived2'.
+                // public class Derived2 : Base { } // 2, 3
+                Diagnostic(ErrorCode.ERR_FileTypeBase, "Derived2").WithArguments("Base", "Derived2").WithLocation(3, 14));
+        }
+
+        [Fact]
+        public void BaseClause_02()
+        {
+            var source = """
+                file interface Interface { }
+
+                class Derived1 : Interface { } // ok
+                file class Derived2 : Interface { } // ok
+
+                interface Derived3 : Interface { } // 1
+                file interface Derived4 : Interface { } // ok
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,11): error CS9301: File type 'Interface' cannot be used as a base type in non-file type 'Derived3'.
+                // interface Derived3 : Interface { } // 1
+                Diagnostic(ErrorCode.ERR_FileTypeBase, "Derived3").WithArguments("Interface", "Derived3").WithLocation(6, 11));
+        }
+
+        [Fact]
+        public void TypeArguments_01()
+        {
+            var source = """
+                file class C { }
+                class Container<T> { }
+                class Program
+                {
+                    Container<C> M() => new Container<C>(); // 1
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,18): error CS9300: File type 'Container<C>' cannot be used in a member signature in non-file type 'Program'.
+                //     Container<C> M() => new Container<C>(); // 1
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M").WithArguments("Container<C>", "Program").WithLocation(5, 18));
+        }
+
+        [Fact]
+        public void Constraints_01()
+        {
+            var source = """
+                file class C { }
+
+                file class D
+                {
+                    void M<T>(T t) where T : C { } // ok
+                }
+
+                class E
+                {
+                    void M<T>(T t) where T : C { } // 1
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (10,30): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'E.M<T>(T)'.
+                //     void M<T>(T t) where T : C { } // 1
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "C").WithArguments("C", "E.M<T>(T)").WithLocation(10, 30));
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Please consider testing `ERR_FileTypeDisallowedInSignature` for:
- record parameters
- lambda return type and parameters
- local function return type and parameters